src/openforms/js/components/admin/form_design/registrations/objectsapi/ObjectsApiOptionsFormFields.js

```diff
@@ -0,0 +1,447 @@
+import {produce} from 'immer';
+import React, {useContext, useState} from 'react';
+import {FormattedMessage, useIntl} from 'react-intl';
+import {TabList, Tabs} from 'react-tabs';
+import useAsync from 'react-use/esm/useAsync';
+
+import {CustomFieldTemplate} from 'components/admin/RJSFWrapper';
+import Tab from 'components/admin/form_design/Tab';
+import {REGISTRATION_OBJECTTYPES_ENDPOINT} from 'components/admin/form_design/constants';
+import {Checkbox, NumberInput, TextArea, TextInput} from 'components/admin/forms/Inputs';
+import Select from 'components/admin/forms/Select';
+import {ValidationErrorContext} from 'components/admin/forms/ValidationErrors';
+import {get} from 'utils/fetch';
+
+import {getErrorMarkup, getFieldErrors} from './utils';
+
+const LOADING_OPTION = [['...', 'loading...']];
+
+const Wrapper = ({children}) => (
+  <form className="rjsf" name="form.registrationBackendOptions">
+    <CustomFieldTemplate displayLabel={false} errors={null}>
+      <fieldset id="root">{children}</fieldset>
+    </CustomFieldTemplate>
+  </form>
+);
+
+const getObjecttypeVersionsEndpoint = objecttype => {
+  const bits = [
+    '/api/v2/registration/plugins/objects-api/object-types',
+    encodeURIComponent(objecttype.uuid),
+    'versions',
+  ];
+  return bits.join('/');
+};
+
+const ObjectsApiOptionsFormFields = ({index, name, schema, formData, onChange}) => {
+  const intl = useIntl();
+  const validationErrors = useContext(ValidationErrorContext);
+
+  const [availableObjecttypes, setAvailableObjecttypes] = useState([]);
+  const [availableObjecttypeVersions, setAvailableObjecttypeVersions] = useState([]);
+
+  const v2SwitchMessage = intl.formatMessage({
+    defaultMessage: `Switching to the new registration options will remove the existing JSON templates.
+    You will also not be able to save the form until the variables are correctly mapped.
+    Are you sure you want to continue?
+    `,
+    description: 'V2 Switch warning message',
```

Comment:
this needs more context - this is the sole description that will be present in the message catalog.

```suggestion
    description: 'Objects API registration backend: v2 switch warning message',
```